Audit fixes

durch opened the pull request:
# Description

Various fixes

# Checklist:

- [ ] added a changelog entry to `CHANGELOG.md`

durch committed:
oak-2

durch committed:
oak-8

durch committed:
oak-13

durch committed:
oak-15

durch committed:
oak-18

durch committed:
Minor clippy nit

jstuczyn commented:
for oak-13 we should put `set_contract_version` in `migrate` function so that it would get populated and updated on every migration. Furthermore we should probably import `cw2` as dependency that has that functionality rather than trying to reimplement it.

jstuczyn commented:
~oak-2 might not be yet fully mitigated due to similar issue as described in: https://gitlab.nymte.ch/nym/security/-/issues/34. Since the `node_identity_signature` is made only on the node's owner, malicious proxy could just replay the signature it has seen when the mixnode bonded (since it would be identical)~

edit: nah, should be fine after all since only the vesting contract is allowed to be the proxy now and given our threat model, the signature replay is not going to happen.

jstuczyn committed:
2023-01-13-OAK-6

jstuczyn committed:
2023-01-13-OAK-3

jstuczyn committed:
2023-01-13-OAK-13

Implemented via direct dependency on cw2 and calling the appropriate code on migration

jstuczyn committed:
Removed few instances of password being unecessarily copied

jstuczyn committed:
2023-01-13-OAK-10

jstuczyn committed:
2023-01-13-OAK-12

jstuczyn committed:
2021-09-13-JP-S-NYM-02

jstuczyn committed:
2021-09-13-JP-S-NYM-03

jstuczyn committed:
Removed further instances of needlessly copying the mnemonic

jstuczyn committed:
2021-09-13-JP-O-PROT-03

jstuczyn committed:
2021-09-13-JP-S-NYM-01*

*: we still have one vulnerability on 'time' pulled from chrono via sqlx. However, apparently its usage is fine... Having said that, I'd still recommend removing all dependencies on chrono, but this will require some database migrations...

jstuczyn committed:
2023-01-13-OAK-11 (#3009)

* wip

* Introducing the concept of starting epoch transition in `nym-api`

* split epoch operations into multiple files

* epoch operation failure recovery

* sending rewarding transactions in correct order

* tests and fixes due to epoch state progression

* lint

jstuczyn committed:
missed rebasing import changes

jstuczyn committed:
Setting cw2 contract version during first migration run

jstuczyn committed:
calling 'reconcile_epoch_events' at least once

jstuczyn committed:
Made message to BeginEpochTransition more consistent with other variants

durch committed:
Merge layer assignment updates
